feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/data/proxy/RealProxySyncService.kt

```diff
@@ -0,0 +1,166 @@
+
+        insertMetaAndChainAccounts(proxiesDiff)
+        insertProxies(proxiesDiff)
+    }
+
+    private suspend fun insertMetaAndChainAccounts(proxiesDiff: CollectionDiffer.Diff<ProxyAccountLocal>) {
+        val identitiesByChain = proxiesDiff.added.loadProxiedIdentities()
+
+        val chainAccounts = proxiesDiff.added.map { proxy ->
+            val identity = identitiesByChain[proxy.chainId]?.get(proxy.proxiedAccountId.intoKey())
+            val proxiedMetaId = accountDao.insertMetaAccountWithNewPosition { nextPosition ->
+                createMetaAccount(proxy.chainId, proxy.proxiedAccountId, identity, nextPosition)
+            }
+            createChainAccount(proxiedMetaId, proxy.chainId, proxy.proxiedAccountId)
+        }
+
+        accountDao.insertChainAccounts(chainAccounts)
+    }
+
+    private suspend fun insertProxies(proxiesDiff: CollectionDiffer.Diff<ProxyAccountLocal>) {
+        val deactivatedProxies = proxiesDiff.removed.map { it.copy(status = ProxyAccountLocal.Status.DEACTIVATED) }
+
+        accountDao.insertProxies(proxiesDiff.newOrUpdated)
+        accountDao.insertProxies(deactivatedProxies)
+    }
+
+    override suspend fun syncForMetaAccount(metaAccount: MetaAccount) {
+        TODO("provide updater to sync proxy delegators for new added accounts")
+    }
+
+    private suspend fun getMetaAccounts(): List<MetaAccount> {
+        return accounRepository.allMetaAccounts()
+            .filter {
+                when (it.type) {
+                    LightMetaAccount.Type.SECRETS,
+                    LightMetaAccount.Type.PARITY_SIGNER,
+                    LightMetaAccount.Type.LEDGER,
+                    LightMetaAccount.Type.POLKADOT_VAULT -> true
+
+                    LightMetaAccount.Type.WATCH_ONLY,
+                    LightMetaAccount.Type.PROXIED -> false
+                }
+            }
+    }
+
+    private suspend fun getSupportedProxyChains(): List<Chain> {
+        return chainRegistry.findChains { it.supportProxy }
+    }
+
+    private suspend fun Chain.getAvailableAccountIds(metaAccounts: List<MetaAccount>): List<MetaAccountId> {
+        return metaAccounts.mapNotNull { metaAccount ->
+            val accountId = metaAccount.accountIdIn(chain = this)
+            accountId?.let {
+                MetaAccountId(accountId, metaAccount.id)
+            }
+        }
+    }
+
+    private suspend fun createMetaAccount(chainId: ChainId, proxiedAccountId: AccountId, identity: Identity?, position: Int): MetaAccountLocal {
+        val chain = chainRegistry.getChain(chainId)
+        return MetaAccountLocal(
+            substratePublicKey = null,
+            substrateCryptoType = null,
+            substrateAccountId = if (chain.isSubstrateBased) proxiedAccountId else null,
+            ethereumPublicKey = null,
+            ethereumAddress = if (chain.isEthereumBased) proxiedAccountId else null,
+            name = identity?.name ?: chain.addressOf(proxiedAccountId),
```

Comment:
I dont think we should fill `substrateAccountId ` and `ethereumAddress ` since proxied wallet only makes sense in one network. So it is better to just fill a single ChainAccount. It will also help us to avoid showing unreletaed balances